scripts/network-init

```diff
@@ -36,24 +36,29 @@ prepare_networking() {
             else
                 PIF="${PHYSICAL_PIF}"
             fi
+            isStatic=false
             case $MODE in
                 dhcp|none)
                     $XE pif-reconfigure-ip uuid=${PIF} mode=${MODE} || true
                     ;;
                 static)
-                    domain=`echo $DOMAIN | tr ' ' ','`
+                    isStatic=true
                     $XE pif-reconfigure-ip uuid=${PIF} mode=static IP=${IP} netmask=${NETMASK} gateway=${GATEWAY} ${DNS+DNS=$DNS} || true
-                    [ $domain ] && $XE pif-param-set uuid=${PIF} other-config:domain=$domain || true
                     ;;
             esac
             case $MODEV6 in
                 dhcp|none|autoconf)
                     $XE pif-reconfigure-ipv6 uuid=${PIF} mode=${MODEV6} || true
                     ;;
                 static)
-                    $XE pif-reconfigure-ipv6 uuid=${PIF} mode=static IPv6=${IPv6} gateway=${IPv6_GATEWAY} || true
+                    isStatic=true
+                    $XE pif-reconfigure-ipv6 uuid=${PIF} mode=static IPv6=${IPv6} gateway=${IPv6_GATEWAY} ${DNS+DNS=$DNS} || true
                     ;;
             esac
+            if [ $isStatic = true ] ; then
```

Comment:
```suggestion
            if $isStatic; then
```

Reply:
I don't agree with this, it makes the check more brittle:
```
$ (export isStatic="" && if $isStatic; then echo yes; else echo no; fi)
yes
```

Keeping the brackets to interpret the variable as a boolean is a good idea in this case.

```
$ (export isStatic="" && if [ $isStatic ]; then echo yes; else echo no; fi)
no
```

Reply:
Maybe just keep the existing @benjamreis if it works, as bash is rather confusing:

```
$ export isStatic=false && if [ $isStatic ]; then echo "true"; fi
true
```

Reply:
I'm not an expert in bash, it's working and from my research it seems to be the standard way to do it. So i'll jsut remove `= true` since `isStatic` is a boolean.
I need to keep the `[]` because the brackets made the check on the value of `isStatic` instead of it definitions.